document-store/src/main/java/org/hypertrace/core/documentstore/parser/IFilteringExpressionParser.java

```diff
@@ -0,0 +1,10 @@
+package org.hypertrace.core.documentstore.parser;
+
+import org.hypertrace.core.documentstore.expression.LogicalExpression;
+import org.hypertrace.core.documentstore.expression.RelationalExpression;
+
+public interface IFilteringExpressionParser {
```

Comment:
It's a personal preference but I prefer not using the prefix "I" for interfaces since interfaces are what gets used widely and it hurts readability. We could just simply use FilteringExpressionParser since the impl class anyway has the db type prefix like MongoFilteringExpressionParser